crates/ntx-builder/src/state/mod.rs

```diff
@@ -235,6 +235,9 @@ impl State {
                 },
             }
 
+            // If this account was in-progress, then it should no longer be as this update is the
+            // result of our own network transaction.
+            self.in_progress.remove(&prefix);
```

Comment:
This mechanism would need to change once we allow transactions affecting network accounts to be submitted externally, right?

Reply:
I _think_ it might be sufficient still, but we should definitely think about the exact mechanics here. On first blush my thoughts are that essentially any local ntx would be racing this external tx and then

- either our local ntx wins and its the same as before, or
- we lose the race and our ntx gets rejected by the block-producer (which already unlocks the account)

So at worst we do one tx's worth of work for now reason. This is already the same scenario as an external tx consuming a note, or something being reverted. The block-producer should remain the "sync" point.